### PR TITLE
#2 Add integration test 

### PR DESCRIPTION
**Pull request by AlexanderScherbatiy**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          # Install other
+          sudo apt-get -y install libudev-dev libusb-1.0-0-dev
+    - run:
+        name: Build platform-contracts
+        command: |
+          cd $SINGNET_REPOS
+          git clone https://github.com/singnet/platform-contracts.git
+          cd platform-contracts
+          npm install
+          npm install ganache-cli
```
Comment (vsbogd) [inline .circleci/config.yml]: Actually commands above doesn't build platform-contracts they just clone repository and install dependencies. To build contracts you should run: ```npm run-script compile``` and ```npm run-script package-npm```.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    - run:
+        name: Build snet-cli
+        command: |
+          cd $SINGNET_REPOS
+          git clone https://github.com/singnet/snet-cli.git
+          cd snet-cli
+          ./scripts/blockchain install
```
Comment (vsbogd) [inline .circleci/config.yml]: I think we need to copy contracts from ```token-contracts``` and ```platform-contracts``` instead

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          # install IPFS
+          wget https://dist.ipfs.io/go-ipfs/v0.4.17/go-ipfs_v0.4.17_linux-amd64.tar.gz
+          tar xvfz go-ipfs_*.tar.gz
+          cp ./go-ipfs/ipfs /usr/local/bin
+          # Installl Python
+          sudo apt-get -y install python3 python3-pip
+          # Install other
+          sudo apt-get -y install libudev-dev libusb-1.0-0-dev
+    - run:
+        name: Build platform-contracts
```
Comment (vsbogd) [inline .circleci/config.yml]: It is worth to add ```token-contracts``` as well.

File: publish_example_service_test.go [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+		return err
+	}
+
+	_, err = check_with_timeout(check_file_contains_strings)
+
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(2 * time.Second)
```
Comment (vsbogd) [inline publish_example_service_test.go]: Why it is needed? Usually you cannot rely on such delays.

File: utils.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+	if env != nil && len(env) > 0 {
+		for _, e := range env {
+			cmd.Env = append(cmd.Env, e)
+		}
+	}
+
+	return cmd, nil
+}
+
+func check_with_timeout(f check_with_timeout_type) (bool, error) {
```
Comment (vsbogd) [inline utils.go]: I would add timeout to the function parameters.

File: utils.go [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+		return err
+	}
+
+	defer file.Close()
+
+	_, err = file.WriteString(content)
+	return err
+}
+
+func link_file(file_from string, file_to string) error {
```
Comment (vsbogd) [inline utils.go]: May be ```make_symlink```?

File: publish_example_service_test.go (hunk 4 above)
Comment (stellarspot) [reply publish_example_service_test.go]: Removing the delay requires adding some logging to the snet-daemon and example service that they are ready. I would prefer to make it as a separate fix.

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+          ./scripts/blockchain install
+          pip3 install -e .
+    - run:
+        name: Build snet-daemon
+        command: |
+          export PATH=$PATH:$GOPATH/bin
+          cd $SINGNET_REPOS
+          git clone https://github.com/singnet/snet-daemon.git
+          cd snet-daemon
+          ./scripts/install
```
[the remark this answers is not in the document]
Comment (stellarspot) [reply .circleci/config.yml]: I update the circleci config to build projects using downloaded parents.

File: utils.go (hunk 5 above)
Comment (stellarspot) [reply utils.go]: The timeout and tick arguments are added to the checkWithTimeout func.

File: utils.go (hunk 6 above)
Comment (stellarspot) [reply utils.go]: Update to use os.Symlink instead of running ln from a command line.

**Pull request merged**